src/content/reference/react/use-server.md

```diff
@@ -24,34 +24,233 @@ canary: true
+
+Notably, these are not supported:
+* React Elements, or [JSX](https://react.dev/learn/writing-markup-with-jsx)
+* Functions, this includes component functions or any other function that is not a server action
+* [Classes](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Classes_in_JavaScript)
+* Objects that are not an instance of [Object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object), or are [null-prototype objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object#null-prototype_objects)
+* Symbols not registered globally, ex. `Symbol('my new symbol')`
+
+
+Supported serializable return values are the same as [serializable props](/reference/react/use-client#passing-props-from-server-to-client-components) for a boundary Client Component.
+
+
+### Security considerations {/*security*/}
 
-* Remember that parameters to functions marked with `'use server'` are fully client-controlled. For security, always treat them as untrusted input, making sure to validate and escape the arguments as appropriate.
-* To avoid the confusion that might result from mixing client- and server-side code in the same file, `'use server'` can only be used in server-side files; the resulting functions can be passed to client components through props.
+Recall that parameters to server actions are fully client-controlled. For security, always treat them as untrusted input, and make sure to validate and escape arguments as appropriate.
+
+In addition, we recommend validating user authorization for any server mutations a server action may perform.
+
+<Wip>
+
+Beyond arguments, server actions also encode closures which are sent to the client and back to when the server action is called. To prevent sending sensitive data, there are experimental taint APIs to prevent unique values and objects from being passed to client code.
+
+See [experimental_taintUniqueValue](/reference/react/experimental_taintUniqueValue) and [experimental_taintObjectReference](/reference/react/experimental_taintObjectReference).
+
+</Wip>
+
+
+#### Caveats {/*caveats*/}
+* `'use server'` can only be used in server-side files; the resulting functions can be passed to Client Components through props.
 * Because the underlying network calls are always asynchronous, `'use server'` can be used only on async functions.
-* Directives like `'use server'` must be at the very beginning of their function or file, above any other code including imports (comments above directives are OK). They must be written with single or double quotes, not backticks. (The `'use xyz'` directive format somewhat resembles the `useXyz()` Hook naming convention, but the similarity is coincidental.)
+* server actions must be called in [transitions](). When used in a `<form action>`, form will ensure the server action is called in a transition. Elsewhere, you will need to manually call a server action in a transition.
```

Comment:
```suggestion
* server actions must be called in [transitions](/reference/react/useTransition). When used in a `<form action>`, form will ensure the server action is called in a transition. Elsewhere, you will need to manually call a server action in a transition.
```

Reply:
can you try another pass at the wording here? a bit hard to parse and to follow

also would be good to accommodate for component libraries doing the transition automatically in the wording